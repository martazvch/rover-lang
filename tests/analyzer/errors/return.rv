return 4

fn main() {}