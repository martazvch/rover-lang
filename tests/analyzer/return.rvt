// Tests return statements

code:
fn add() { return }
fn sub() -> int { return 5 }
fn mul(a: int) -> float { return a * sub() }
fn main() {}

expect:
[Fn declaration add, index: 0, scope: Global, return kind: Explicit]
    [Return expr: false, cast: false]
[Fn declaration sub, index: 1, scope: Global, return kind: Explicit]
    [Return expr: true, cast: false]
        [Int 5]
[Fn declaration mul, index: 2, scope: Global, return kind: Explicit]
    [Return expr: true, cast: true]
        [Binop type: MulInt, cast: None]
            [Variable index: 1, scope: Local]
            [Fn call arity: 0, builtin: false]
                [Variable index: 1, scope: Global]
        [Cast to Float]
[Fn declaration main, index: 3, scope: Global, return kind: ImplicitVoid]

================

code:
return 4

fn main() {}

error:
ReturnOutsideFn

================

