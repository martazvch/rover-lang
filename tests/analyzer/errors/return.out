return_outside_fn
